modules/library/metadata/src/main/java/org/geotools/util/Utilities.java

```diff
@@ -670,4 +672,16 @@ public static void ensureNonNull(final String name, final Object object)
         }
     }
 
+    // TODO: document
+    public static <T> Stream<T> stream(Optional<T> optional) {
```

Comment:
we will of course need javadocs for public methods

Reply:
I just documented the utility methods, but would really appreciate if someone else took a shot at the `FactoryRegistry` methods. I would mostly be guessing about the details...